modules/consensus/accept_test.go

```diff
@@ -27,60 +36,248 @@ func (db mockDbTx) Bucket(name []byte) dbBucket {
 	return db.buckets[string(name)]
 }
 
+// predefinedBlockUnmarshal is a predefined response from mockBlockMarshaler. It
+// defines the unmarshaled processedBlock and error code that mockBlockMarshaler
+// should return in response to an input serialized byte slice.
+type predefinedBlockUnmarshal struct {
+	serialized  []byte
+	unmarshaled processedBlock
+	err         error
+}
+
+// mockBlockMarshaler is an implementation of the encoding.GenericMarshaler
+// interface for unit testing. It allows clients to specify mappings of
+// serialized bytes into unmarshaled blocks.
+type mockBlockMarshaler struct {
+	p []predefinedBlockUnmarshal
+}
+
+// Marshal is not implemented and panics if called.
+func (m mockBlockMarshaler) Marshal(interface{}) []byte {
+	panic("not implemented")
+}
+
+// Unmarshal unmarshals a byte slice into an object based on a pre-defined map
+// of deserialized objects.
+func (m mockBlockMarshaler) Unmarshal(b []byte, v interface{}) error {
+	for _, pu := range m.p {
+		if bytes.Equal(b[:], pu.serialized[:]) {
+			pv, ok := v.(*processedBlock)
+			if !ok {
+				panic("mockBlockMarshaler.Unmarshal expected v to be of type processedBlock")
+			}
+			*pv = pu.unmarshaled
+			return pu.err
+		}
+	}
+	panic("unmarshal failed: predefined unmarshal not found")
+}
+
+// AddPredefinedUnmarshal adds a predefinedBlockUnmarshal to mockBlockMarshaler.
+func (m *mockBlockMarshaler) AddPredefinedUnmarshal(u predefinedBlockUnmarshal) {
+	m.p = append(m.p, u)
+}
+
+// mockBlockRuleHelper is an implementation of the blockRuleHelper interface for
+// unit testing.
+type mockBlockRuleHelper struct {
+	minTimestamp types.Timestamp
+}
```

Comment:
Fixed, didn't create a `mock_block.go` as I wasn't sure what would make sense to go there, since we're not creating a mock for `types.block` but open to suggestions